File: Arrays/ChocolateDist.js
const chocolateDistribute = (chocolates, students, numOfChocolates) => {
  if (!chocolates || !students) return -1;
  if (chocolates < students) return -1;

  chocolates = chocolates.sort((a, b) => a - b);
  //   console.log(chocolates);
  let minDifference = Number.MAX_VALUE;

  for (let i = 0; i + students - 1 < numOfChocolates; i++) {
    if (chocolates[i + students - 1] - chocolates[i] < minDifference) {
      minDifference = chocolates[i + students - 1] - chocolates[i];
      //   console.log(minDifference);
    }
  }
  return minDifference;
};

const arr = [12, 4, 7, 9, 2, 23, 25, 41, 30, 40, 28, 42, 30, 44, 48, 43, 50];
console.log("Minimum Difference: ", chocolateDistribute(arr, 7, arr.length));

File: Arrays/ChocolateDist.java

public class ChocolateDist {

    public static int chocolateDistribute(int[] arr, int m) {
        if (arr.length <= 0 || m == 0 || arr.length - 1 < m)
            return -1;

        int minDifference = Integer.MAX_VALUE;
        for (int i = 0; i < arr.length; i++) {

            int window = i + m - 1;
            if (window > arr.length - 1)
                break;

            int diff = arr[window] - arr[i];

            if (diff < minDifference)
                minDifference = diff;
        }
        return minDifference;

    }

    public static void main(String[] args) {
        int arr[] = { 12, 4, 7, 9, 2, 23, 25, 41, 30, 40, 28, 42, 30, 44, 48, 43, 50 };
        int m = 7;

        // Calculate the result using the chocolateDistribution method
        int result = chocolateDistribute(arr, m);

        // Print the result or indicate an invalid input
        if (result != -1) {
            System.out.println("Minimum difference is " + result);
        } else {
            System.out.println("Invalid input");
        }

    }
}
